include/mxnet/ndarray.h

```diff
@@ -771,31 +773,31 @@ class NDArray {
    * This function returns dnnl::memory with the given primitive_desc
    * as long as the array size meets the required size in the given primitive_desc.
    */
-  const dnnl::memory* GetDNNLData(const dnnl::memory::desc& md) const;
+  const dnnl::memory* GetDNNLData(const void* md) const;
   /*
    * This function returns dnnl::memory with the given primitive_desc.
    * The returned dnnl::memory will have the same physical layout as
    * the given primitive_desc.
    */
-  const dnnl::memory* GetDNNLDataReorder(const dnnl::memory::desc& md) const;
+  const dnnl::memory* GetDNNLDataReorder(const void* md) const;
 
   /*
    * This function copies data from dnnl memory.
    */
-  void CopyFrom(const dnnl::memory& mem);
+  void CopyFrom(const void* mem);
```

Comment:
```suggestion
  void CopyFrom(const dnnl::memory* mem);
```
I guess using declared above dnnl::memory should make oneDNN implementation not contaminated with many static_cast later (?)